Headers/DebugServer2/Utils/Bits.h

```diff
@@ -0,0 +1,88 @@
+//
+// Copyright (c) 2014-present, Facebook, Inc.
+// All rights reserved.
+//
+// This source code is licensed under the University of Illinois/NCSA Open
+// Source License found in the LICENSE file in the root directory of this
+// source tree. An additional grant of patent rights can be found in the
+// PATENTS file in the same directory.
+//
+
+#ifndef __DebugServer2_Utils_Bits_h
+#define __DebugServer2_Utils_Bits_h
+
+#include <type_traits>
+
+#include "DebugServer2/Utils/CompilerSupport.h"
+
+namespace ds2 {
+namespace Utils {
+namespace {
+// We apply the builtin bit functions on chunks of sizeof(int) * 8 bits.
+// However, we can't do full shifts of this size because only the least
+// signifitive bits are used and clang complains, so we shift twice half
+// the size.
+const size_t kHalfIntBits = sizeof(int) * 4;
```

Comment:
Because that way, `kHalfIntBits` is local to this translation unit and
might be elided at compilation. Not marking it `static` forces the compiler
to emit a symbol for it (because it can be used by other translation units).

On Thu, Mar 10, 2016 at 11:15 AM, walter erquinigo <notifications@github.com
> wrote:

> In Headers/DebugServer2/Utils/Bits.h
> <https://github.com/facebook/ds2/pull/211#discussion_r55733368>:
>
> > +
> > +#ifndef __DebugServer2_Utils_Bits_h
> > +#define __DebugServer2_Utils_Bits_h
> > +
> > +#include <type_traits>
> > +
> > +#include "DebugServer2/Utils/CompilerSupport.h"
> > +
> > +namespace ds2 {
> > +namespace Utils {
> > +namespace {
> > +// We apply the builtin bit functions on chunks of sizeof(int) * 8 bits.
> > +// However, we can't do full shifts of this size because only the least
> > +// signifitive bits are used and clang complains, so we shift twice half
> > +// the size.
> > +const size_t kHalfIntBits = sizeof(int) * 4;
>
> teach me why, please
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/facebook/ds2/pull/211/files#r55733368>.
>



-- 
Stephane Sezer